### PR TITLE
Add route and component for calculating social circle

### PR DESCRIPTION
**Pull request by jasontyu**
* Added new route [GET `/api/users/:secret/exposure`] for getting count of social circle size
  * Added UserService function `getSocialCircle(secret)` for getting all userIds of friends in social circle
  * Added helper UserModel functions `getUserIdBySecret`, `getFriendsById`

* Added new Component `CountDisplay` for displaying exposure count and handling state. 
  * Imported/rendered in `AddFriend`

### CODE REVIEW

File: routes/api.js [hunk 1]
```diff
@@ -39,6 +39,15 @@ api.get('/user/:secret', function (req, res) {
     })
 })
 
+// Route for getting exposure count for user
+api.get('/user/:secret/exposure', async function (req, res) {
```
Comment (LeekSwan) [inline routes/api.js:43]: I thought the route was `/api${this.props.location.pathname}/exposure`

File: client/src/AddFriend.js [hunk 2]
```diff
@@ -118,8 +120,7 @@ class AddFriend extends React.Component {
           {displayAlert(this.state)}
         </div>
 
-        <h5>Total Count</h5>
-        <h5>0</h5>
+        <CountDisplay location={this.props.location}/>
```
Comment (LeekSwan) [inline client/src/AddFriend.js:123]: What does location refer to?

Comment (jasontyu) [reply client/src/AddFriend.js:123]: It's used for `location.pathname` to get the base URL for the API call, including the user secret. You can ctrl+f for it in CountDisplay

File: routes/api.js (hunk 1 above)
Comment (jasontyu) [reply routes/api.js:43]: > start.js
Screen Shot 2020-07-22 at 12 38 05 PM

When the request is forwarded to the API router, the `/api` part is chopped off. Or to think of it the other way around, the route definition has `/api` pre-pended to it.

**Pull request merged**